pkg/asset/installconfig/azure/validation.go

```diff
@@ -995,3 +1000,65 @@ func checkBootDiagnosticsURI(client API, diag *aztypes.BootDiagnostics, region s
 	}
 	return nil
 }
+
+func validateSubnetNatGateway(client API, fieldPath *field.Path, subnet *aznetwork.Subnet, outboundType aztypes.OutboundType, role capz.SubnetRole, resourceGroup, virtualNetwork string) field.ErrorList {
+	var allErrs field.ErrorList
+	if outboundType != aztypes.NATGatewayMultiZoneOutboundType && outboundType != aztypes.NATGatewaySingleZoneOutboundType {
+		return allErrs
+	}
+	if role == capz.SubnetControlPlane || virtualNetwork == "" || resourceGroup == "" {
+		return allErrs
```

Comment:
Ditto: allErrs doesn't have any errors in it at this point.

Reply:
Same here. Must run only if the subnet role is compute.

Reply:
It should be fine... I think. The real problem is I'm not able to attach a NAT gateway here if they specify Multi zone outbound type.

Reply:
I'll remove it.